atlasdb-commons/src/main/java/com/palantir/common/collect/IterableView.java

```diff
@@ -212,6 +215,23 @@ public T getFirst() {
         return IteratorUtils.wrap(delegate().iterator());
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public Spliterator<T> spliterator() {
+        return (Spliterator<T>) delegate().spliterator();
+    }
+
+    /**
+     * @return a sequential {@code Stream} with the contents as its source.
+     */
+    @SuppressWarnings("unchecked")
+    public Stream<T> stream() {
+        if (delegate() instanceof Collection) {
+            return ((Collection<T>) delegate()).stream();
+        }
+        return StreamSupport.stream(spliterator(), false);
```

Comment:
Given we have Guava here, please can we just do `return Streams.stream(delegate())` which is functionally equivalent?

Reply:
I _think_ your unchecked casts are safe, having gone through all the cases. Annoyingly the ones below in castAsList and castAsCollection are not, since

```
List<Long> longs = Lists.newArrayList(1L);
List<Number> numbers = IterableView.of(longs).castAsList();
numbers.add(1.0);
assertThat(longs.stream().allMatch(Long.class::isInstance)).isTrue();
```

will sadly compile.

Do you fancy quickly changing them to wrap with Collections.unmodifiableCollection and Collections.unmodifiableList while you're modifying this class?

Reply:
bugs beget bugs etc; I'm ok with classes suppressing warnings iff every usage in that class is safe :)

Reply:
lame, wish guava had a better type here (streams aren't mutable so you should be able to make a ? extends T a stream of T